src/code_execution/godbolt.rs

```diff
@@ -51,34 +56,188 @@ struct GodboltTool {
     stderr: GodboltOutput,
 }
 
+#[derive(Debug, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct GodboltTarget {
+    id: String,
+    name: String,
+    lang: String,
+    compiler_type: String,
+    semver: String,
+    instruction_set: String,
+}
+
+impl GodboltTarget {
+    fn clean_request_data(&mut self) {
+        // Some semvers get weird characters like `()` in them or spaces, we strip that out here
+        self.semver = self
+            .semver
+            .chars()
+            .filter(|char| char.is_alphanumeric() || matches!(char, '.' | '-' | '_'))
+            .map(|char| char.to_ascii_lowercase())
+            .collect();
+    }
+}
+
+async fn update_godbolt_targets(data: &Data) -> Result<PoolConnection<Sqlite>, Error> {
+    let mut conn = data.database.acquire().await?;
+
+    // Fetch the last time we updated the targets list, this will be null/none if we've never
+    // done it before
+    let last_update_time = sqlx::query!("SELECT last_update FROM last_godbolt_update")
+        .fetch_optional(&mut conn)
+        .await?;
+
+    let needs_update = if let Some(last_update_time) = last_update_time {
+        // Convert the stored timestamp into a utc date time
+        let last_update_time = Utc.timestamp_nanos(last_update_time.last_update);
+
+        // Get the time to wait between each update of the godbolt targets list
+        let update_period = chrono::Duration::from_std(
+            env::var("GODBOLT_UPDATE_DURATION")
```

Comment:
The environment variable parsing could be done in main() instead and stored in Data. I think that'd be nicer because other environment variables are already parsed in main() too, plus the command function has less non-business logic